### PR TITLE
DM-40287: Package cleanups

### PR DESCRIPTION
**Pull request by timj**
* Fix location of SqlRegistry
* Add numpydoc and ruff fixes
* Switch to pyproject.toml
* Fix mypy
* Update dependencies

### CODE REVIEW

File: .github/workflows/docstyle.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  call-workflow:
+    uses: lsst/rubin_workflows/.github/workflows/docstyle.yaml@main
+    with:
+      args: "$(find python -name '*.py' | egrep -v 'parser/ply|parserYacc')"
```
Comment (andy-slac) [inline .github/workflows/docstyle.yaml:13]: This looks like copy from daf_butler, probably not needed here?

File: python/lsst/dax/obscore/obscore_exporter.py [hunk 2]
```diff
@@ -135,7 +141,18 @@ def __init__(self, fname: str, null_value: bytes, sep_in: bytes, sep_out: bytes)
...
+        """Write next buffer to output.
+
+        Parameters
+        ----------
+        buffer : `bytes`
+            Buffer to write.
+
+        Returns
+        -------
+        `int`
```
Comment (andy-slac) [inline python/lsst/dax/obscore/obscore_exporter.py:153]:
```suggestion
        size: `int`
```

**Review by andy-slac — APPROVED** [review]
Thanks for all fixes! Looks great, a couple of minor comments.

File: python/lsst/dax/obscore/obscore_exporter.py (hunk 2 above)
Comment (timj) [reply python/lsst/dax/obscore/obscore_exporter.py]: Ironically numpydoc itself prefers not to have a variable name if there is only one thing being returned. It's the Rubin style guide that insists on the name. Fixed.

**Pull request merged**